runtime/memprof.c

```diff
@@ -223,7 +236,10 @@ static value do_callback(tag_t tag, intnat wosize, int32_t occurrences,
                          value callstack, enum ml_alloc_kind cb_kind) {
   CAMLparam1(callstack);
   CAMLlocal1(sample_info);
-  CAMLassert(occurrences > 0);
+  CAMLassert(occurrences > 0 && !caml_memprof_suspended);
+  value res; /* Not a root, can be an exception result. */
```

Comment:
`msvc64` complains about declaring a variable after an assert instruction; see the [CI build failure log](https://ci.appveyor.com/project/avsm/ocaml/builds/24760806/job/pg67sp03rlbt4dd5).